### ISSUE TITLE
Fetching values at intersections of anchors in GInteractions?

### ISSUE THREAD

**Issue by kajocina** [liz-is/readhic#1]
Hi!

Sorry for writing here, but I can't see a more direct contact link plus it might be useful to others.
I downloaded your code for parsing Rao HiC data and have the data stored as InteractionSet. I also have a list of pairs of genomic positions and wanted to extract their supporting normalized read values from the Rao data.

Do you know how can I extract these values for my genomic pairs? I read the vignettes of InteractionSet and GInteractions but the examples there are using other types of input data. I can't look at how the parsed Rao data looks like in the InteractionSet because the print will never work, only crash my system. Therefore I can't even try doing it "by hand".

So in short, I wanted to obtain data with such columns:

`anchor1_granges     anchor2_granges    normalized_read_count
`

Best,
Piotr

**Comment by liz-is** [comment]
That's fine, I think this is a good place for questions - as you said, it could be useful to others :)

If your pairs of genomic positions are stored as a GInteractions object (which can be constructed from 2 GRanges objects, for example), you can do 2D overlaps between that object and the HiC data.
e.g.

```
findOverlaps(ints_of_interest, all_hic)
```

This will give you the indices of interactions in the HiC data which overlap each of your pairs. Alternatively, you might find the `linkOverlaps` function useful if you are interested in all possible interactions between two sets of GRanges that aren't paired.

You might prefer to store the data as a GInteractions object with the interaction counts in metadata columns. This loses some of the possibilities of the InteractionSet object but is a simpler structure and has a printing method that will show you the data you want. You can do this by:

```
new_gi <- interactions(all_hic) #extract GInteractions object
new_gi$RAWobserved <- as.vector(assays(all_hic)[["RAWobserved"]]) #only works if there is a single column in the assay (i.e. a single sample)
```

I might add a convenience function to do this for all the assays in the InteractionSet object, if you'd find that useful.

You can also access the assay values for a subset of interactions like this:

```
ints_of_interest <- c(1, 2, 100, 112)
assays(all_hic[ints_of_interest])[["RAWobserved"]]
```

I hope this helps, please ask if anything isn't explained well enough. I haven't had any trouble printing InteractionSet objects so not sure what the problem is there, sorry.

**Comment by kajocina** [comment]
Great, thank you for your help!

I think I am fine with typing in names of the assays in the list subset by hand.
As I understand, having used the KRnorm normalization vector and after loading the data using your code with normalization performed, I should fetch both RAWexpected and RAWobserved values and calculate the O/E ratio, right?

**Comment by liz-is** [comment]
The normalisation function adds the normalised data as new assays, the original data isn't modified. You should fetch the KRnorm and KRexpected assays (or the equivalent for another normalisation method if you prefer).

**Comment by kajocina** [comment]
Just to be double-sure: after calling the normalization function with a selected method (KR in my case), the added KRnorm vector is already the Observed/Expected ratios using normalized data? Or the Obs/Exp would then be formed by dividing KRnorm / KRexpected vectors?

**Comment by liz-is** [comment]
The KRnorm vector is the normalised observed data. The observed / expected
ratio is not automatically calculated. You should use the appropriate
expected vector for the normalisation you use, I.e. KRnorm and KRexpected
in this case.

**Comment by kajocina** [comment]
Great, thank you for your help. I'm quite new to HiC but seems to me that for a person having some experience using GRanges it's not a steep learning curve to find your way around in GenomicInteractions. What a powerful way to work with huge data ;)

**Comment by liz-is** [comment]
I'm glad you're finding it not too hard to use! Can you confirm that this answered your question so I can close this issue?

**Comment by kajocina** [comment]
Yes, all is clear, your code helped me get quicker to some interesting results. Have a nice day!
